tests/check/check.go

```diff
@@ -1206,45 +1225,136 @@ condition xcond(x: string) {
 					t.Skipf("no check assertions defined")
 				}
 				for _, assertion := range stage.CheckAssertions {
-					t.Run(fmt.Sprintf("assertion_%s", assertion.Name), func(t *testing.T) {
-						detailedInfo := fmt.Sprintf("Check request: %s. Tuples: %s. Contextual tuples: %s", assertion.Tuple, stage.Tuples, assertion.ContextualTuples)
-
-						var tupleKey *openfgav1.CheckRequestTupleKey
-						if assertion.Tuple != nil {
-							tupleKey = &openfgav1.CheckRequestTupleKey{
-								User:     assertion.Tuple.GetUser(),
-								Relation: assertion.Tuple.GetRelation(),
-								Object:   assertion.Tuple.GetObject(),
-							}
-						}
-						resp, err := client.Check(ctx, &openfgav1.CheckRequest{
-							StoreId:              storeID,
-							AuthorizationModelId: modelID,
-							TupleKey:             tupleKey,
-							ContextualTuples: &openfgav1.ContextualTupleKeys{
-								// TODO
-								TupleKeys: []*openfgav1.TupleKey{},
-							},
-							Context: assertion.Context,
-							Trace:   true,
-						})
-
-						if assertion.ErrorCode == 0 {
-							require.NoError(t, err, detailedInfo)
-							require.Equal(t, assertion.Expectation, resp.GetAllowed(), detailedInfo)
-						} else {
-							require.Error(t, err, detailedInfo)
-							e, ok := status.FromError(err)
-							require.True(t, ok, detailedInfo)
-							require.Equal(t, assertion.ErrorCode, int(e.Code()), detailedInfo)
-						}
+					t.Run(fmt.Sprintf("assertion_check_%s", assertion.Name), func(t *testing.T) {
+						assertCheck(ctx, t, assertion, stage, client, storeID, modelID)
+					})
+					t.Run(fmt.Sprintf("assertion_list_objects_%s", assertion.Name), func(t *testing.T) {
+						assertListObjects(ctx, t, assertion, client, storeID, modelID)
+					})
+					t.Run(fmt.Sprintf("assertion_list_users_%s", assertion.Name), func(t *testing.T) {
+						assertListUsers(ctx, t, assertion, client, storeID, modelID)
 					})
 				}
 			})
 		}
 	})
 }
 
+func assertCheck(ctx context.Context, t *testing.T, assertion *checktest.Assertion, stage *stage, client ClientInterface, storeID string, modelID string) {
+	detailedInfo := fmt.Sprintf("Check request: %s. Tuples: %s. Contextual tuples: %s", assertion.Tuple, stage.Tuples, assertion.ContextualTuples)
+
+	var tupleKey *openfgav1.CheckRequestTupleKey
+	if assertion.Tuple != nil {
+		tupleKey = &openfgav1.CheckRequestTupleKey{
+			User:     assertion.Tuple.GetUser(),
+			Relation: assertion.Tuple.GetRelation(),
+			Object:   assertion.Tuple.GetObject(),
+		}
+	}
+	resp, err := client.Check(ctx, &openfgav1.CheckRequest{
+		StoreId:              storeID,
+		AuthorizationModelId: modelID,
+		TupleKey:             tupleKey,
+		ContextualTuples: &openfgav1.ContextualTupleKeys{
+			// TODO
```

Comment:
TODO what ? :)   (I know is to have test for the context tuple keys case )

Reply:
Yes hehe